checks.d/mongo.py

```diff
@@ -354,49 +352,52 @@ def __init__(self, name, init_config, agentConfig, instances=None):
     def get_library_versions(self):
         return {"pymongo": pymongo.version}
 
-    def check_last_state(self, state, clean_server_name, agentConfig):
-        if self._last_state_by_server.get(clean_server_name, -1) != state:
-            self._last_state_by_server[clean_server_name] = state
-            return self.create_event(state, clean_server_name, agentConfig)
-
-    def create_event(self, state, clean_server_name, agentConfig):
+    def check_last_state(self, state, clean_server_name, replSet, agentConfig):
+        last_state = self._last_state_by_server.get(clean_server_name, -1)
+        self._last_state_by_server[clean_server_name] = state
+        if last_state != state and last_state != -1:
+            return self.create_event(last_state, state, clean_server_name, replSet['set'], agentConfig)
+
+    def hostname_for_event(self, clean_server_name, agentConfig):
+        """Return a reasonable hostname for a replset membership event to mention."""
+        uri = urlsplit(clean_server_name)
+        hostname = uri.netloc.split(':')[0]
+        if hostname == 'localhost':
+            hostname = get_hostname(agentConfig)
```

Comment:
The result is the same but when the hostname is not specified in `datadog.conf` file, the `get_hostname` method has to re-resolve it, when the value is already accessible, stored in an instance variable, https://github.com/DataDog/dd-agent/blob/master/checks/__init__.py#L326